test/e2e/create_test.go

```diff
@@ -262,16 +332,37 @@ func (tc *testContext) waitForWindowsNodes(nodeCount int32, waitForAnnotations,
 					node.Annotations[nodeconfig.PubKeyHashAnnotation], pubKeyAnnotation)
 				return false, nil
 			}
+			// At this point it is confident that we are looking at fully configured nodes only, so we can check if a
+			// node is from BYOH or not. This check cannot be made earlier as the BYOH annotation is only going to be
+			// applied at the end of configuration
+			if node.Annotations[controllers.BYOHAnnotation] == "true" {
+				if isBYOH {
+					windowsNodes = append(windowsNodes, node)
+				}
+			} else {
+				if !isBYOH {
+					windowsNodes = append(windowsNodes, node)
+				}
+			}
```

Comment:
I still think this check is strange. Can't you first check `isBYOH` and only then check for the BYOH annotation?

Reply:
Only other way I could see this refactored is making the clause a one-liner
```
if (isBYOH && node.Annotations[controllers.BYOHAnnotation] == "true") || 
		(!isBYOH && node.Annotations[controllers.BYOHAnnotation] != "true") {
	windowsNodes = append(windowsNodes, node)
}
```